8.gtex-interpret/scripts/nbconverted/3.apply-signatures.py

```diff
@@ -0,0 +1,501 @@
+
+# coding: utf-8
+
+# # Application of Compressed Gene Expression Signatures
+# 
+# **Gregory Way, 2018**
+# 
+# We previously identified a specific VAE feature (z = 3) that captured blood signatures.
+# These signatures were not captured in VAE z = 2, and, when present, contributed to a rapid increase in the ability to capture the signal in GTEX blood tissues.
+# 
+# The primary differences between the two VAE models appeared to be related to neutrophil and monocyte signatures. Here, we test the ability of these signatures to generalize to external datasets.
+# 
+# 
+# ## Part 1:
+# 
+# ### Enrichment of Neutrophil Signatures
+# 
+# Here, we apply the VAE feature enriched for neutrophil genes to a publicly available dataset capturing neutrophil differentiation in two leukemia cell lines.
+# 
+# ![cell_type_Neutrophils_HPCA_2.png](https://github.com/greenelab/BioBombe/raw/master/6.analyze-weights/figures/GTEX/signal/GpXCELL/gene_set_Neutrophils_HPCA_2.png)
+# 
+# ## Part 2:
+# 
+# ### Enrichment of Monocyte Signatures
+# 
+# Here, we apply the VAE features enriched for monocyte genes to a different publicly available dataset that captures various cell-types undergoing hematopoiesis.
+# 
+# ![cell_type_Monocytes_FANTOM_2.png](https://github.com/greenelab/BioBombe/raw/master/6.analyze-weights/figures/GTEX/signal/GpXCELL/gene_set_Monocytes_FANTOM_2.png)
+# 
+# ## Output
+# 
+# In both cases, various scores are output that will be visualized in a separate notebook.
+
+# In[1]:
+
+
+import os
+import sys
+import numpy as np
+import pandas as pd
+import matplotlib.pyplot as plt
+import seaborn as sns
+
+import hetio.readwrite
+import hetmech.matrix
+
+from scripts.utils import load_weight_matrix, apply_signature, load_enrichment_results
+
+sys.path.append('../scripts')
+from latent import parse_gmt
+
+
+# In[2]:
+
+
+get_ipython().run_line_magic('matplotlib', 'inline')
+
+
+# In[3]:
+
+
+np.random.seed(123)
+
+
+# ## 1.0. Load and Process External Neutrophil Dataset
+
+# In[4]:
+
+
+file = os.path.join('data', 'GSE103706_processed_matrix.tsv.gz')
+geo_scaled_zeroone_df = pd.read_table(file, index_col=0)
+
+print(geo_scaled_zeroone_df.shape)
+geo_scaled_zeroone_df.head(2)
+
+
+# ## 1.1. Apply Signature from VAE z = 3 (feature 0)
+
+# In[5]:
+
+
+vae_z3_seed = 908341
+vae_z3_feature = 'vae_0'
+
+
+# In[6]:
+
+
+weight_z3_df = load_weight_matrix(dataset='GTEX',
+                                  z_dim=3,
+                                  seed=vae_z3_seed)
+
+result_vae_3_feat0 = apply_signature(weight_df=weight_z3_df,
+                                     other_df=geo_scaled_zeroone_df,
+                                     feature=vae_z3_feature,
+                                     align=True)
+
+
+# ## 1.2. Apply Signature from VAE z = 14 (feature 10)
```

Comment:
We wanted to test the feature that displayed the highest enrichment for `Neutrophils_HPCA_2`. Again, definitely confusing and will add comments!

![gene_set_neutrophils_hpca_2](https://user-images.githubusercontent.com/7340421/50912789-f5542a00-1400-11e9-9fa9-7e6ccfeb55c8.png)

Also, gonna try to avoid hardcoding the seed and feature variables too